File: day4/0_threads.rb
def display_words(word_array)
  20.times do |n|
    print "#{n}: "
    word_array.each { |word| print "#{word} " }
    print "\n"
  end
end

days = %w(Monday Tuesday Wednesday Thursday Friday Saturday Sunday)
months = %w(Jan Feb Mar Apr May Jun Jul Aug Sep Oct Nov Dec)

t1 = Thread.new { display_words(days) }
t2 = Thread.new { display_words(months) }

t1.join
t2.join

File: day4/0_threads.py
import threading

def display_words(word_array):
    for n in range(20):
        print(f"{n}: ", end="")
        for word in word_array:
            print(f"{word} ", end="")
        print()  # newline

days = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"]
months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]

# Create threads
t1 = threading.Thread(target=display_words, args=(days,))
t2 = threading.Thread(target=display_words, args=(months,))

# Start and join threads
t1.start()
t2.start()
t1.join()
t2.join()